eduvpn/nm.py

```diff
@@ -293,12 +293,19 @@ def start_wireguard_connection(
 
     dns4 = []
     dns6 = []
-    for ip in config['Interface']['DNS'].split(','):
-        address = ip_address(ip.strip())
-        if address.version == 4:
-            dns4.append(str(address))
-        elif address.version == 6:
-            dns6.append(str(address))
+    dns_hostnames = []
+    for dns_entry in config['Interface']['DNS'].split(','):
+        stripped_entry = dns_entry.strip()
+        try:
+            address = ip_address(stripped_entry)
+            if address.version == 4:
+                dns4.append(str(address))
+            elif address.version == 6:
+                dns6.append(str(address))
+        # The entry is not an ip but a hostname
+        # They need to be added to dns search domains
+        except ValueError:
```

Comment:
I prefer to keep the try branch as short as possible, with an `else` branch for the rest.